src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -1608,6 +1635,14 @@ def delete_flavor(self, flavor_id: UUID) -> None:
             except NoResultFound as error:
                 raise KeyError from error
 
+    def _purge_non_custom_flavors(self) -> None:
+        """Delete all non-custom flavors."""
+        with Session(self.engine) as session:
+            non_custom_flavors = session.exec(
+                select(FlavorSchema).where(FlavorSchema.is_custom is False)
```

Comment:
I think we had a similar problem with the filter model queries where we used `is` instead of sqlalchemy `.is_(...)`. This might also be a problem here?

Reply:
just to be safe I will switch to .is_()

Reply:
So apparently you need to go with is_( when you are using the following API:

`getattr(table, "workspace_id").is_(None)`